ion-schema/src/isl/isl_range.rs

```diff
@@ -944,6 +976,23 @@ impl<T: Display> Display for RangeBoundaryValue<T> {
     }
 }
 
+impl<T: Display> WriteToIsl for RangeBoundaryValue<T> {
+    fn write_to<W: IonWriter>(&self, writer: &mut W) -> IonSchemaResult<()> {
+        match &self {
+            Max => writer.write_symbol("max")?,
+            Min => writer.write_symbol("min")?,
+            Value(value, range_boundary_type) => {
+                if range_boundary_type == &RangeBoundaryType::Exclusive {
+                    writer.set_annotations(["exclusive"]);
+                }
+                let element = Element::read_one(format!("{value}").as_bytes())?;
```

Comment:
I don't have a concrete suggestion at the moment, but this bit feels a bit fragile to me. We're using the constraint `T: Display` but actually hoping that `T` is something that's serialized as text Ion, which is a narrower set of types.